lib/calendar.js

```diff
@@ -0,0 +1,72 @@
+import Simulator from './simulator-xcode-6';
+import path from 'path';
+import { fs, mkdirp } from 'appium-support';
+import { execSQLiteQuery } from './utils';
+
+let Calendar;
+
+async function getCalendarDB (sharedResourcesDir) {
+  let tccPath = path.resolve(sharedResourcesDir, 'Library', 'TCC');
+  if (!(await fs.exists(tccPath))) {
+    await mkdirp(tccPath);
+  }
+  
+  let db = path.resolve(tccPath, 'TCC.db');
+  await execSQLiteQuery(db, `CREATE TABLE IF NOT EXISTS access (
+    service TEXT NOT NULL DEFAULT '', 
+    client TEXT NOT NULL DEFAULT '', 
+    client_type INTEGER,
+    allowed INTEGER,
+    prompt_count INTEGER,
+    csreq BLOB NOT NULL DEFAULT '', 
+    policy_ID INTEGER, 
+    PRIMARY KEY(service, client, client_type)
+  );`.replace(/\n/g, ''));
+
+  return db;
+}
+
+async function getCalendarRowCount (db, bundleID) {
+  let count = await execSQLiteQuery(db, `SELECT count(*) FROM access WHERE client='?';`, bundleID);
+  return parseInt(count.stdout.split('=')[1], 10);
+}
+
+/**
+ * Allow calendar access for UDID
+ * @param {string} udid 
+ * @param {string} bundleID 
+ */
+async function enableCalendarAccess (udid, bundleID) {
```

Comment:
Since `enableCalendarAccess` and `disableCalendarAccess` are both taking the `udid` of a simulator, and do their business by creating that simulator object and getting the path from it, it seems to me that these functions ought to be on the simulator object itself. The driver that is trying to enable or disable the calendar has access to the simulator object, so the [calling code](https://github.com/appium/appium-xcuitest-driver/blob/9e6b357a2e49ac2276b71e4c4a07e7e0411c7053/lib/driver.js#L329-L338) goes from
```js
    if (!this.isRealDevice()) {
      // Enable calendar access if calendarAccessAuthorized flag is set
      // Disable calendar access if it is explicitly set to false
      if (this.opts.calendarAccessAuthorized) {
        enableCalendarAccess(this.opts.udid, this.opts.bundleId);
      } else if (this.opts.calendarAccessAuthorized === false) {
        disableCalendarAccess(this.opts.udid, this.opts.bundleId);
      }
    }
```
to
```js
    if (!this.isRealDevice()) {
      // Enable calendar access if calendarAccessAuthorized flag is set
      // Disable calendar access if it is explicitly set to false
      if (this.opts.calendarAccessAuthorized) {
        this.opts.device.enableCalendarAccess(this.opts.bundleId);
      } else if (this.opts.calendarAccessAuthorized === false) {
        this.opts.device.disableCalendarAccess(this.opts.bundleId);
      }
    }
```